draft-irtf-cfrg-hpke.md

```diff
@@ -1207,6 +1207,34 @@ across all labels in this document.
 The `0xFFFF` AEAD ID is reserved for applications which only use the Export
 interface; see {{hpke-export}} for more details.
 
+# API Considerations {#api-considerations}
+
+The high-level HPKE APIs specified in this document are all fallible.
+For example, `Decap()` can fail if the encapsulated key `enc` is invalid,
+and `Open()` may fail if ciphertext decryption fails. The explicit errors
+generated throughout this specification, along with the conditions that
+lead to each error, are as follows:
+
+- `ValidationError`: Public key validation failure; {{dhkem}}.
+- `DeserializeError`: Public or private key deserialization failure; {{base-crypto}}.
+- `DecapError`: `Decap()` failure; {{base-crypto}}.
+- `OpenError`: Context AEAD `Open()` failure; {{base-crypto}} and {{hpke-dem}}.
+- `MessageLimitReachedError`: Context AEAD sequence number overflow; {{base-crypto}} and {{hpke-dem}}.
+
+Implicit errors may also occur. As an example, certain classes of failures,
+e.g., malformed recipient public keys, may not yield explicit errors.
+For example, for the DHKEM variant described in this specification,
+the `Encap()` algorithm may fail when given an invalid recipient public key.
+However, other KEM algorithms may not have an efficient algorithm for verifying
+the validity of public keys. As a result, an equivalent error may not manifest
+until AEAD decryption at the recipient. As another example, DHKEM's `AuthDecap()`
+function will produce invalid output if given the wrong sender public key.
+This error is not detectable until subsequent AEAD decryption.
+
+Applications using HPKE APIs should not assume that classes of failures,
+especially those that stem from implicit errors, will always manifest the
+same way. Applications SHOULD account for errors when encountered.
```

Comment:
This is super vague.  What are you actual actually trying to say here?

Reply:
As in the PR description, I'm trying to say "don't expect only certain types of errors in certain types of places -- be prepared to handle errors anywhere." It would be a shame if an API couldn't adopt a new ciphersuite because their error handling doesn't permit it, for example.